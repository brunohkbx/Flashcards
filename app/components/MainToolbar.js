import * as React from 'react';
import PropTypes from 'prop-types';
import { Toolbar, ToolbarContent, ToolbarAction } from 'react-native-paper';

const MainToolbar = ({ handleSearchPress }) => {
  return (
    <Toolbar>
      <ToolbarContent title="Flashcards" />
      <ToolbarAction icon="search" onPress={handleSearchPress} />
    </Toolbar>
  );
};

MainToolbar.propTypes = {
  handleSearchPress: PropTypes.func.isRequired
};

export default MainToolbar;
