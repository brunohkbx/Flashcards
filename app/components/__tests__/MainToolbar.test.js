import React from 'react';
import { shallow } from 'enzyme';
import MainToolbar from '../MainToolbar';

describe('MainToolbar', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      handleSearchPress: jest.fn()
    }, propOverrides)

    const wrapper = shallow(<MainToolbar {...defaultProps} />);

    const pressSearchIcon = () => {
      wrapper
        .find('ToolbarAction')
        .simulate('press')
    }

    return {
      wrapper,
      pressSearchIcon
    }
  }

  it('renders properly', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  test('SearchIcon onPress calls handleSearchPress', () => {
    const mockhandleSearchPress = jest.fn();
    const { pressSearchIcon } = setup({ handleSearchPress: mockhandleSearchPress });

    pressSearchIcon();

    expect(mockhandleSearchPress).toHaveBeenCalled();
  });
});
