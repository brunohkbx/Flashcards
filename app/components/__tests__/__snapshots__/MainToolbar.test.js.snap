// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainToolbar renders properly 1`] = `
<withTheme(Toolbar)>
  <withTheme(ToolbarContent)
    title="Flashcards"
  />
  <ToolbarAction
    icon="search"
    onPress={[MockFunction]}
    size={24}
  />
</withTheme(Toolbar)>
`;
